ktor-network/jvm/src/io/ktor/network/sockets/JavaSocketAddress.kt

```diff
@@ -0,0 +1,118 @@
+/*
+ * Copyright 2014-2022 JetBrains s.r.o and contributors. Use of this source code is governed by the Apache 2.0 license.
+ */
+
+package io.ktor.network.sockets
+
+import java.lang.reflect.*
+
+public actual sealed class SocketAddress {
+    internal abstract val address: java.net.SocketAddress
+}
+
+public actual class InetSocketAddress internal constructor(
+    override val address: java.net.InetSocketAddress
+) : SocketAddress() {
+
+    // May trigger a name service reverse lookup when called.
```

Comment:
Please consider writing a regular KDoc(`/** */`) with docs

Reply:
@e5l added some documentation. What do you think about it?

Reply:
@andreyaksenov, could you check?

Reply:
@e5l looks good